File: rgctl-tests/ecommerce-python/app/routers/categories.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.middleware.auth import require_auth
from app.schemas.category import CategoryResponse, CreateCategoryRequest
from app.services.category import CategoryService
from app.utils.dependencies import AuthUser

router = APIRouter(prefix="/api/categories", tags=["categories"])


@router.get("", response_model=list[CategoryResponse])
def list_categories(db: Session = Depends(get_db)) -> list[CategoryResponse]:
    return CategoryService(db).list_all()


@router.get("/{category_id}", response_model=CategoryResponse)
def get_category(category_id: str, db: Session = Depends(get_db)) -> CategoryResponse:
    return CategoryService(db).get(category_id)


@router.post("", response_model=CategoryResponse, status_code=201)
def create_category(
    req: CreateCategoryRequest,
    _user: AuthUser = Depends(require_auth),
    db: Session = Depends(get_db),
) -> CategoryResponse:
    return CategoryService(db).create(req)

File: rgctl-tests/ecommerce-typescript/src/routes/categories.ts
import { Router } from 'express';
import { requireAuth } from '../middleware/auth';
import { asyncHandler } from '../middleware/errorHandler';
import * as categoryService from '../services/categoryService';

const router = Router();

router.get(
  '/',
  asyncHandler(async (_req, res) => {
    res.json(categoryService.listCategories());
  }),
);

router.post(
  '/',
  requireAuth,
  asyncHandler(async (req, res) => {
    const result = categoryService.createCategory(req.body);
    res.status(201).json(result);
  }),
);

router.get(
  '/:id',
  asyncHandler(async (req, res) => {
    const result = categoryService.getCategory(String(req.params.id));
    res.json(result);
  }),
);

export default router;
